### ISSUE TITLE
Browserstack support

### ISSUE THREAD

**Issue by naxhh** [sitespeedio/browsertime#288]
Hi everyone.

I want to make this project compatible with Browserstack.

But I don't know the project too deep. So my main question is. What options/features I should check?

Right now my current QA will be just running in from my browser and comparing results from a browserstack run.

But there are some flags I should keep care of?
For example, video recording is one I detected and I need to see how it works.

Any advice on this will be much appreciated!

**Comment by soulgalore** [comment]
Hey @naxhh 
you could probably hack in username/password somewhere [here](https://github.com/sitespeedio/browsertime/blob/fcf6b7231381c7b6e76b65a01eae8c7cce6b407c/lib/core/webdriver/index.js) to just try.

Still I think stability/connectivity will be an issue as I said in
https://github.com/sitespeedio/sitespeed.io/issues/1495

The video uses FFMPeg for a local run. For Browserstack we will need to implement downloading the video etc. Not sure though, there are [some steps](https://github.com/sitespeedio/browsertime/tree/fcf6b7231381c7b6e76b65a01eae8c7cce6b407c/lib/support/video) that needs to work in Browserstack.

Best
Peter

**Comment by naxhh** [comment]
Yes. I have runs working agains browserstack. But as I don't know the full project maybe I miss some feature that needs adapting, video is just a good example :)

Ok so I'll start hacking agains the video and as soon as I have a PR send it and see what happens :)

**Comment by soulgalore** [comment]
cool, however to merge/add Browserstack support I want to feel confident that the metrics are stable and that they can have some kind of connectivity setup. When I used Browsertack for Selenium tests I haven't had the best experience.

**Comment by naxhh** [comment]
Sure, I understand if the metrics are not stable enough or the connection options can't be set it's not worth having support for it.

**Comment by soulgalore** [comment]
Have you had time to test and what did you find out @naxhh ?

**Comment by naxhh** [comment]
Hi. I've performed a feck quick checks regarding connecting to Browser stack and performing the tests there.

The part I didn't find out is how to retrieve the video, and without that most of the metrics are not there.
We ended up deploying the sitespeed tool into an amazon machine with docker. Because that approach was faster and provided us with some benefits.

So I stopped looking into this at that point.
I don't know if you want me to close this issue or leave it open for future efforts? Either way is fine for me. And thanks for your guidance in my tests.
